### PR TITLE
af-packet: Remove support for rollover 

### PR DESCRIPTION
**Pull request by jlucovsky**
Continuation of #9069 
This MR removes support for AF-Packet rollover.

Rollover support is defined [here](https://www.man7.org/linux/man-pages/man7/packet.7.html).

If `cluster-type: cluster_rollover` is configured, a warning message will be displayed and the clustering used will be `cluster_flow`.

The warning message:
```
Warning: af-packet: enp6s0f0: cluster_rollover deprecated; using "cluster_flow" instead. See https://redmine.openinfosecfoundation.org/issues/6128
```
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: [6128](https://redmine.openinfosecfoundation.org/issues/6128)

Describe changes:
- Document deprecation in upgrade notes
- Detect when rollover is used (`cluster-type: cluster_rollover`) and use `cluster_flow` instead

Updates
- Removed `rollover` handling per review feedback.
- 
### Provide values to any of the below to override the defaults.

To use a pull request use a branch name like `pr/N` where `N` is the pull request number.

```
SV_REPO=
SV_BRANCH=
SU_REPO=
SU_BRANCH=
LIBHTP_REPO=
LIBHTP_BRANCH=
```

### CODE REVIEW

**Comment by suricata-qa** [comment]
WARNING:
 
 | field | baseline | test | % |
|--|--|--|--| 
| | SURI_TLPR1_stats_chk | | |
| .flow.memuse | 592373632 | 520052840 | 87.79% | 

Pipeline 14954

**Review by victorjulien — CHANGES_REQUESTED** [review]
This still leaves the regular `rollover` option. I think that should just turn into a no-op with a warning.

File: src/runmode-af-packet.c [hunk 1]
```diff
@@ -375,7 +375,12 @@ static void *ParseAFPConfig(const char *iface)
         SCLogConfig("%s: using round-robin cluster mode for AF_PACKET", aconf->iface);
         aconf->cluster_type = PACKET_FANOUT_LB;
         cluster_type = PACKET_FANOUT_LB;
-    } else if (strcmp(tmpctype, "cluster_flow") == 0) {
+    } else if (strcmp(tmpctype, "cluster_flow") == 0 || strcmp(tmpctype, "cluster_rollover") == 0) {
+        if (strcmp(tmpctype, "cluster_rollover") == 0) {
+            SCLogWarning("%s: cluster_rollover deprecated; using \"cluster_flow\" instead. See "
+                         "https://redmine.openinfosecfoundation.org/issues/6128",
```
Comment (victorjulien) [inline src/runmode-af-packet.c:381]: I think I prefer: "See ticket #6128."

I think we use this ticket notation elsewhere too.

File: suricata.yaml.in [hunk 2]
```diff
@@ -621,6 +621,7 @@ af-packet:
     #  more info.
     # Recommended modes are cluster_flow on most boxes and cluster_cpu or cluster_qm on system
     # with capture card using RSS (requires cpu affinity tuning and system IRQ tuning)
+    # cluster_rollover has been deprecated; if used, it'll be replaced with cluster_flow.
```
Comment (victorjulien) [inline suricata.yaml.in:624]: I read this as "in the future, it will be replaced by ...". Maybe reword a bit... "this now acts as cluster_flow" or something like this?

**Comment by jlucovsky** [comment]
Continued in #9148
